### PR TITLE
Fixed bug in spade, linked to missing call of min_neu, added test

### PR DESCRIPTION
**Pull request by pbouss**
Resolving issue: https://github.com/NeuralEnsemble/elephant/issues/574

### CODE REVIEW

File: elephant/spade.py [hunk 1]
```diff
@@ -912,6 +913,7 @@ def _fpgrowth(transactions, min_c=2, min_z=2, max_z=None,
                 report='a',
                 algo='s',
                 winlen=winlen,
+                min_neu=min_neu,
```
Comment (Moritz-Alexander-Kern) [inline elephant/spade.py:916]: This line appears to be the relevant change if I understand correctly. It seems to address the core issue and provide a solution. Thanks!

Please correct me if I'm wrong about this.

File: elephant/test/test_spade.py [hunk 2]
```diff
@@ -348,6 +348,41 @@ def test_fpgrowth_fca(self):
...
+        bin_size = 5 * pq.ms
+
+        spade_output = spade.spade(
+            spiketrains,
+            bin_size=bin_size,
+
+            # +1 to pattern length to be able to find it again
+            winlen=2,
+            min_spikes=n_spiketrains,
+            max_spikes=n_spiketrains,
```
Comment (Moritz-Alexander-Kern) [inline elephant/test/test_spade.py:370]: Is the usage of `n_spiketrains` in this context specifically due to its value being 3? Or does the number of spiketrains have a direct impact on `min_spikes` and `max_spikes`?

File: elephant/test/test_spade.py [hunk 3]
```diff
@@ -348,6 +348,41 @@ def test_fpgrowth_fca(self):
...
+            spectrum='3d#')
+
+        patterns = spade_output['patterns']
+        assert len(patterns) == 1
+        pattern = patterns[0]
+        assert len(np.unique(pattern['neurons'])) == n_spiketrains
+        assert_array_equal(pattern['itemset'], [2, 5, 1])
+        assert_array_equal(pattern['windows_ids'], [36, 134, 856])
+        assert_array_equal(pattern['neurons'], [1, 2, 0])
+        assert_array_equal(pattern['signature'], [3, 3, 1])
```
Comment (Moritz-Alexander-Kern) [inline elephant/test/test_spade.py:384]: I'm curious to know why exactly this pattern is expected or considered the correct one. Are there any previous results or a rationale behind this choice? 
I noticed that the spiketrains were generated using the `StationaryPoissonProcess` class, so perhaps there is an obvious reason that I'm not aware of. 
I'd appreciate any insights or explanations to further educate myself on this matter.

**Review by Moritz-Alexander-Kern — COMMENTED** [review]
Thanks for taking care of this bug and the code cleanup @pbouss .
**Cleanup**
- replaced old string formatting with f-strings
- sorted imports
- removed `_fpgrowth_filter` function (was already commented out)

**Bugfix**
- see Line [916](https://github.com/NeuralEnsemble/elephant/pull/575/files#diff-f086abec9c91a6cf36d2b8009c4d662000c81736514adc5e9758f1e99411d6dbR916): hand over `min_neu` as argument to `fim.fpgrowth` - function.
- implemented unit-test to check for expected pattern

In conclusion the issue was not with the `fim.so` but with the way arguments were passed to it? Please correct me if I get this wrong.

**Comment by pbouss** [comment]
The old `fim.so`, i.e. the Borgelt version didn't have the argument. Therefore, there was this extra cleaning function that was outcommented when including the Porrman FIM implementation, which works correctly. - It was just forgotten to pass the argument.

File: elephant/test/test_spade.py (hunk 2 above)
Comment (pbouss) [reply elephant/test/test_spade.py:370]: I chose arbitrarily the 3 here. I just wanted to have a case, where I can have an unwanted autopatterns, that is with a repeating neuron. - This I get with this Poisson spike trains and choice of parameters.

File: elephant/test/test_spade.py (hunk 3 above)
Comment (pbouss) [reply elephant/test/test_spade.py:384]: There is no specific reason. Originally I only had this line: 
```
for pattern in patterns:
    # Check that all spiketrains contain three different neurons
    assert len(np.unique(pattern['neurons']))==n_spiketrains
```

Then, I thought for later code changes it might help to fix here a specific reproducible outcome. But there is no specific reason that the pattern looks like this. It's only the outcome given the random seed.

**Review by Moritz-Alexander-Kern — COMMENTED** [review]
build wheels
